Allow players to start a game with fewer than 3 hearts

Aelire opened the pull request:
Currently the minimum number of heart containers to start a seed with is set to 3, matching vanilla. As a difficulty option, letting the player start with fewer heart containers adds some difficulty.
This changes the `starting_hcs` option to be the absolute number of starting heart containers, and sets the minimum to 1 (but leaves the default at 3) so that it's possible to start with 2 or 1 heart containers instead of 3

The only bug I ran into while testing that so far is that when saving and quitting, the current health was always reset to 3 regardless of the max number of hearts. I addressed that by piggybacking on the patch to refill the magic meter on save load, adding a cap to the current health so it doesn't go over the max

Finally to avoid incessant beeping, there is a tweak included that prevents beeping at full health. Otherwise when the maximum health is 1 heart the game gives the low health beep permanently.
However, I haven't done anything to the link "tired" animation/hitbox that is active at low health, since it's not as annoying as the beeping (and I have no idea how to disable it)

Things I have tested: 
* starting with 1 heart (no beeping)
* save+quit with 1 heart (resets to 1 heart, not 3)
* lose 0.5 heart with 1 heart (fast beep)
* start with 1 heart, obtain heart container, lose 0.5 heart (slow beep). Lose 1 heart: medium beep
* start with >3 hearts, lose health down to 2 hearts, save+quit (reset to 3 hearts)
* start with 1 heart and hero mode
* Collect 1 heart piece, gameover/continue and save+quit (reset to 1 heart not 1.25)

Aelire commented on wwr_ui/randomizer_window.py:
```diff
@@ -112,6 +112,7 @@ def __init__(self, cmd_line_args):
     self.set_option_description(None)
     
     self.update_settings()
+    self.update_health_label()
```
I noticed that on startup the UI always showed "Current Starting Health: 3 Hearts" regardless of the value that is actually in the `settings.txt` file, this is a small fix for that

LagoLunatic commented:
There's a bug where if you get a game over and choose to continue, your health will be reset to 3 hearts, exceeding the maximum.

Natolumin commented on asm/patches/remove_low_health_beep.asm:
```diff
@@ -0,0 +1,59 @@
+  ; Ignore beep if max hearts == 1
+  lis     r5, 0x803C4C09@ha ; max health offset.
+  addi    r5, r5, 0x803C4C09@l
+  lbz     r6, 0 (r5)
```
Ah thanks, I mistakenly took them to be the same as magic, I'll update these! It does makes a lot more sense that they're s16 starting at the beginning of the object

Aelire commented:
Added the same kind of patch to the gameover continue code too, thanks for catching that! 
I also updated the addresses to offset from the start of `g_dComIfG_gameInfo` instead of pointing in the middle of it

Aelire commented:
Found another bug, when checking / resetting against `maxLife` it needs to be rounded to the closest full heart, otherwise if you have 1 heart container and 1 piece of heart on reset you'll have 1 and 1/4 heart, but the maximum should still be 1 heart until 4 pieces are collected
